Fix: Date-related NewGRF string codes (0x82 and 0x83) should interpret their argument as unsigned.

diff -r 81221bdf30d5 src/newgrf_text.cpp
--- a/src/newgrf_text.cpp	Sat Aug 20 05:35:03 2011 +0000
+++ b/src/newgrf_text.cpp	Sat Aug 20 14:18:13 2011 +0200
@@ -1028,7 +1028,7 @@
 			case SCC_NEWGRF_PRINT_UNSIGNED_WORD:  *argv = _newgrf_textrefstack.PopUnsignedWord();  break;
 
 			case SCC_NEWGRF_PRINT_DATE:
-			case SCC_NEWGRF_PRINT_MONTH_YEAR:     *argv = _newgrf_textrefstack.PopSignedWord() + DAYS_TILL_ORIGINAL_BASE_YEAR; break;
+			case SCC_NEWGRF_PRINT_MONTH_YEAR:     *argv = _newgrf_textrefstack.PopUnsignedWord() + DAYS_TILL_ORIGINAL_BASE_YEAR; break;
 
 			case SCC_NEWGRF_DISCARD_WORD:         _newgrf_textrefstack.PopUnsignedWord(); break;
 
